### PR TITLE
NIFI-8332 Service account impersonation support for GCP credential co…

### PR DESCRIPTION
**Pull request by s9514171**
…ntroller service

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

The GCPCredentialsControllerService already supports service account private key or Application Default Credentials (ADC) to access GCP
But in some use cases/security control, it's necessary to use impersonation to access the service

This PR used to support the service account impersonation for GCP credentials

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [X] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/CredentialPropertyDescriptors.java [hunk 1]
```diff
@@ -89,4 +89,57 @@ private CredentialPropertyDescriptors() {}
...
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .required(true)
+            .sensitive(false)
+            .description("A comma-separated list of scopes to request during the authorization grant. "
+                    + "This property is valid only when using the service account impersonation strategy; otherwise, it is ignored.")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(IMPERSONATE_SERVICE_ACCOUNT)
+            .build();
+
+    public static final PropertyDescriptor IMPERSONATE_CREDENTIAL_LIFETIME = new PropertyDescriptor.Builder()
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/CredentialPropertyDescriptors.java:133]: What will happen when the processor is running for a period over the value defined in this property? Won't the access be revoked?

Comment (s9514171) [reply nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/CredentialPropertyDescriptors.java:133]: Google's SDK will auto-refreshed by itself. if you want to revoke, need to set in IAM

**Comment by s9514171** [comment]
I add the default value for lifetime, the value comes from the [document](https://googleapis.dev/java/google-auth-library/latest/com/google/auth/oauth2/ImpersonatedCredentials.html#create-com.google.auth.oauth2.GoogleCredentials-java.lang.String-java.util.List-java.util.List-int-com.google.auth.http.HttpTransportFactory-)

**Comment by s9514171** [comment]
Hi @pvillard31 
sorry for ping you, does this pull request need any changes
it's running on our servers for a while and runs well

**Pull request closed without merging**